pkg/reconciler/route/config/domain.go

```diff
@@ -63,33 +69,50 @@ func (s *LabelSelector) Matches(labels map[string]string) bool {
 // Domain maps domains to routes by matching the domain's
 // label selectors to the route's labels.
 type Domain struct {
-	// Domains map from domain to label selector.  If a route has
-	// labels matching a particular selector, it will use the
-	// corresponding domain.  If multiple selectors match, we choose
-	// the most specific selector.
-	Domains map[string]*LabelSelector
+	// Domains map from domain to a set of options including a label selector.
+	Domains map[string]DomainConfig
+}
+
+// The configuration of one domain
+type DomainConfig struct {
+	// The label selector for the domain. If a route has labels matching a particular selector, it
+	// will use the corresponding domain. If multiple selectors match, we choose the most specific
+	// selector.
+	Selector *LabelSelector `json:"selector,omitempty"`
+	// If true, the domain will have a wildcard TLS certificate generated.
+	Wildcard bool `json:"wildcard"`
```

Comment:
I think I'd prefer a non-bool field here - maybe `type: Wildcard`?

Unsure what other domains types we might have in the future but it would provide us an extension point.

Reply:
Yeah I could see that, I'm also not sure what sort of other types you might have, but it would give flexibility.